### ISSUE TITLE
`az_heat` only returns data to `date - 2`

### ISSUE THREAD

**Issue by jeremylweiss** [uace-azmet/azmetr#63]
It should return data to `date - 1`. I suspect it involves code in `parse_params.R` that builds the URL of the data request for the API.

For example, at present 
```r 
az_heat("az07", start_date = "2023-10-29", end_date = "2023-10-30")
```
returns a `tbl_df` with `tbl_df$datetime_last` = "2023-10-29"

So, it effectively builds a URL of:
`https://api.azmet.arizona.edu/v1/observations/hueto/az07/2023-10-29T00:00/P1DT0H`
when we'd want this, instead:
`https://api.azmet.arizona.edu/v1/observations/hueto/az07/2023-10-29T00:00/P1DT24H`

The URL:
`https://api.azmet.arizona.edu/v1/observations/hueto/az07/2023-10-29T00:00/P1D`
which *I think* is what the code actually builds, returns the same data as the former URL, above. Do we need to specify "T0H" at the end of the currently constructed URL?

Note: I'm looking at code from the `date-input-flex` branch.

**Comment by Aariq** [comment]
It doesn't appeary to have anything to do with whether time is specified or not.  `P2D` also gets data with `"2023-10-30"` for `datetime_last`.  I think this is more likely a quirk with how dates are interpreted by the API.  I'll do a little investigating and open a PR with a proposed fix.

**Comment by Aariq** [comment]
I think we can fix this by always rounding the end date up to 00:00 of the next day.

**Comment by Aariq** [comment]
I think this was technically fixed by #62 but #64 adds tests for this.
